src/bin.ts

```diff
@@ -1,16 +1,6 @@
 import {Channel, COLUMN, ROW, SHAPE, SIZE} from './channel';
 
-/**
- * Binning properties or boolean flag for determining whether to bin data or not.
- */
-export interface Bin {
-  /**
-   * A two-element (`[min, max]`) array indicating the range of desired bin values.
-   * @minItems 2
-   * @maxItems 2
-   */
-  extent?: number[];
-
+export interface AbstractBin {
```

Comment:
Why AbstractBin shouldn't have `extent`?  (Please add comment to explain)

Also, we use `Base` prefix rather than `Abstract` for the rest of the project.